core/stop/residual_norm.cpp

```diff
@@ -62,6 +66,154 @@ GKO_REGISTER_OPERATION(implicit_residual_norm,
 }  // namespace implicit_residual_norm
 
 
+template <typename ValueType>
+bool any_is_complex()
+{
+    return false;
+}
+
+
+template <typename ValueType, typename LinOp, typename... Rest>
+bool any_is_complex(const LinOp* in, Rest&&... rest)
+{
+#if GINKGO_BUILD_MPI
+    bool is_complex_distributed =
+        dynamic_cast<const ConvertibleTo<distributed::Vector<>>*>(in);
+#else
+    bool is_complex_distributed = false;
```

Comment:
In terms of naming, doesn't this rather check that it's `real`, because you can convert it to real?